src/util/atomicbool.go

```diff
@@ -1,32 +1,35 @@
 package util
 
-import "sync"
+import (
+	"sync/atomic"
+)
 
 // AtomicBool is a boxed-class that provides synchronized access to the
 // underlying boolean value
 type AtomicBool struct {
-	mutex sync.Mutex
-	state bool
+	state int32 // "1" is true, "0" is false
 }
 
 // NewAtomicBool returns a new AtomicBool
 func NewAtomicBool(initialState bool) *AtomicBool {
-	return &AtomicBool{
-		mutex: sync.Mutex{},
-		state: initialState}
+	var state int32 = 0
```

Comment:
Can be simplified:

```go
var state int32
if initialState {
    state = 1
}
```

Reply:
Yes, we can simplify this. But in this particular case I prefer to explicitly set variable to `0`, so that it would be more obvious that `"1" is true` and `"0" is false`